### PR TITLE
Work around issues with black material surfaces

### PR DESCRIPTION
**Pull request by randallfrank**
When possible, use the JSON description of the material to improve on the simpler protobuffer color scheme.

### CODE REVIEW

File: src/ansys/pyensight/core/utils/dsg_server.py [hunk 1]
```diff
@@ -55,6 +68,63 @@ def reset(self, cmd: Any = None) -> None:
         if cmd is not None:
             self.hash.update(cmd.hash.encode("utf-8"))
         self.cmd = cmd
+        self._material = None
+
+    def _parse_material(self) -> None:
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/dsg_server.py:73]: Here I parse the JSON from the protocol buffer to get more material details.  We do support the notion that the JSON can contain multiple representations for different scheme (e.g. MDL vs EnSight vs ...)

File: src/ansys/pyensight/core/utils/omniverse_dsg_server.py [hunk 2]
```diff
@@ -510,16 +521,19 @@ def create_dsg_points(
         return part_stage_url
 
     def create_dsg_material(
-        self, stage, mesh, root_name, diffuse=[1.0, 1.0, 1.0, 1.0], variable=None
+        self, stage, mesh, root_name, diffuse=[1.0, 1.0, 1.0, 1.0], variable=None, mat_info={}
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/omniverse_dsg_server.py:524]: If the JSON material representation is provided, use values in the resulting material dictionary to override the core protobuffer colors/etc.

**Review by mariostieriansys — DISMISSED** [review]

File: src/ansys/pyensight/core/utils/dsg_server.py [hunk 3]
```diff
@@ -55,6 +68,63 @@ def reset(self, cmd: Any = None) -> None:
...
+        Parse the JSON string in the part command material string and
+        make the content accessible via material_names() and material().
+        """
+        if self._material is not None:
+            return
+        try:
+            if self.cmd.material_name:  # type: ignore
+                self._material = json.loads(self.cmd.material_name)  # type: ignore
+                for key, value in self._material.items():
+                    value["name"] = key
```
Comment (david-bremer) [inline src/ansys/pyensight/core/utils/dsg_server.py:84]: What does this do?

Comment (david-bremer) [reply src/ansys/pyensight/core/utils/dsg_server.py:84]: I think I see.  You know the value is a dict, and it was useful to put the name into the dict.

**Review by david-bremer — APPROVED** [review]
I applied 8 metallic materials to 8 rocky shapes, and they came through where one would expect, for all of the shapes defined by a mesh.  They look a little different, but the base color differs, and the shininess comes through.

The color is not applied to the spheres, which use some other shader, so that is expected.

**Comment by randallfrank** [comment]
> I applied 8 metallic materials to 8 rocky shapes, and they came through where one would expect, for all of the shapes defined by a mesh. They look a little different, but the base color differs, and the shininess comes through.
> 
> The color is not applied to the spheres, which use some other shader, so that is expected.

Your call @david-bremer if you want to handle spheres the same way.  I only did triangles and went out of my way to avoid lines (on purpose).

**Pull request merged**